src/extension/messages/apiRequest/createCellExecution.ts

```diff
@@ -0,0 +1,43 @@
+import { ClientMessages } from '../../../constants'
+import { ClientMessage, IApiMessage } from '../../../types'
+import { InitializeClient } from '../../api/client'
+import { getCellByUuId } from '../../cell'
+import { getTerminalByCell } from '../../utils'
+import { createCellExecutionQuery } from '../../api/grapql'
+import { postClientMessage } from '../../../utils/messaging'
+
+type APIRequestMessage = IApiMessage<ClientMessage<ClientMessages.apiRequest>>
+
+export default async function createCellExecution(requestMessage: APIRequestMessage): Promise<void | boolean> {
+    const { messaging, message, editor } = requestMessage
+
+    try {
+
+        const cell = await getCellByUuId({ editor, uuid: message.output.uuid })
+        if (!cell) {
+            throw new Error('Cell not found')
+        }
+        const terminal = getTerminalByCell(cell)
+        if (!terminal) {
+            throw new Error('Could not find an associated terminal')
+        }
+        const pid = await terminal.processId || 0
+        const exitCode = cell.executionSummary?.success ? 0 : -1
```

Comment:
@mxsdev  any ideas on how I can get the exact status code in case of failure, currently, I'm not sending the proper status code, not needed for now, just need to know if succeeded or no. executionSummary does the work.

Reply:
Thanks, added support, non-interactive not supported atm, will do as a separate PR